### DISCUSSION TITLE
Object reference not set to an instance of an object.

### DISCUSSION THREAD

**Discussion by Shrimpiee** [neos-sdi/adfsmfa#252] [Q&A] [answered]
I'm trying to setup MFA TOTP for our domain users on a web application. The site redirect well to adfs and adfs does show the MFA wizard that request the code, but I got the error "Object reference not set to an instance of an object" when enter the code. I'm using Google Authenticator. I don't know if the error was on MFA side or the web application side.

Also when I try to send the key through email, i got an error in the event viewer "Error Sending Email (KEY) for user, with status code GeneralFailure : Failure sending mail".

I don't have much idea since pretty much everything is green and ADFS does detected and using MFA on Access Control Policies. 

Edit: Nevermind, just a simple restart fix both problem...

**Comment by redhook62** [comment]
hi, @Shrimpiee 

It is possible that you have not filled in the various necessary properties, 
For example for sending emails you have a button to test.
Without going into detail, your information is a little too brief. 
I suggest you send us your configuration using **PowerShell: Export-MFASystemConfiguration** and forward the resulting file to you at adfsmfa@gmail.com.
We advise you to re-read the entire Wiki. the number of parameters is important, it can help you to configure the plugin according to your wishes.

regards

**Comment by Shrimpiee** [comment, marked as answer]
Thank you for your answer. I have followed the wiki and try to test each function before posting here. In the end it was just that the ADFS primary server needs a restart, after that, both TOTP and Email are working fine.

Cheers!

**Reply by redhook62** [reply]
Great !
